frontend/src/components/authentication/RegistrationForm.jsx

```diff
@@ -74,64 +75,83 @@ class RegistrationForm extends Component {
     this.setState({
-      isFirstLoad: false,
-      firstNameContent: updatedFirstNameValue,
-      isValidFirstName: isValid
+      firstNameContent: firstNameContent
     });
   };
 
-  lastNameValidation = event => {
-    const updatedLastNameValue = event.target.value;
-    const isValid = validateName(updatedLastNameValue);
+  getLastNameContent = event => {
+    const lastNameContent = event.target.value;
     this.setState({
-      isFirstLoad: false,
-      lastNameContent: updatedLastNameValue,
-      isValidLastName: isValid
+      lastNameContent: lastNameContent
     });
   };
 
-  emailValidation = event => {
-    const updatedEmailValue = event.target.value;
-    const isValid = validateEmail(updatedEmailValue);
-    this.setState({ isFirstLoad: false, emailContent: updatedEmailValue, isValidEmail: isValid });
+  getEmailContent = event => {
+    const emailContent = event.target.value;
+    this.setState({ emailContent: emailContent });
   };
 
-  passwordValidation = event => {
-    const updatedPasswordValue = event.target.value;
-    const passwordConfirmationValue = this.state.passwordConfirmationContent;
-    const isValid = validatePassword(updatedPasswordValue);
+  getPasswordContent = event => {
+    const passwordContent = event.target.value;
     this.setState({
-      isFirstLoad: false,
-      isFirstPasswordLoad: false,
-      passwordContent: updatedPasswordValue,
-      isValidPassword: isValid,
-      isValidPasswordConfirmation: updatedPasswordValue === passwordConfirmationValue
+      passwordContent: passwordContent
     });
   };
 
-  passwordConfirmationValidation = event => {
-    const updatedPasswordConfirmationValue = event.target.value;
-    const passwordValue = this.state.passwordContent;
+  getPasswordConfirmationContent = event => {
+    const passwordConfirmationContent = event.target.value;
+    this.setState({
+      passwordConfirmationContent: passwordConfirmationContent
+    });
+  };
+
+  validateForm = () => {
+    const isValidFirstName = validateName(this.state.firstNameContent);
+    const isValidLastName = validateName(this.state.lastNameContent);
+    const isValidEmail = validateEmail(this.state.emailContent);
+    const isValidPassword = validatePassword(this.state.passwordContent);
+    const passwordContent = this.state.passwordContent;
+    const passwordConfirmationContent = this.state.passwordConfirmationContent;
     this.setState({
       isFirstLoad: false,
-      passwordConfirmationContent: updatedPasswordConfirmationValue,
-      isValidPasswordConfirmation: updatedPasswordConfirmationValue === passwordValue
+      isFirstPasswordLoad: false,
+      accountExistsError: false,
+      isValidFirstName: isValidFirstName,
+      isValidLastName: isValidLastName,
+      isValidEmail: isValidEmail,
+      isValidPassword: isValidPassword,
+      isValidPasswordConfirmation: passwordContent === passwordConfirmationContent
     });
   };
 
+  isFormValid = () => {
```

Comment:
This function can be written as:

```
  isFormValid = () => {
     return this.state.isValidFirstName &&
      this.state.isValidLastName &&
      this.state.isValidEmail &&
      this.state.isValidPassword &&
      this.state.isValidPasswordConfirmation;
  };
```

Reply:
Simplified in commit #https://github.com/code-for-canada/project-thundercat/pull/321/commits/b6bcd249f50afbe07858e378b969d529dac87cf4